ProjectLighthouse.Servers.Website/Controllers/Moderator/ModerationSlotController.cs

```diff
@@ -68,4 +68,24 @@ public async Task<IActionResult> DeleteLevel([FromRoute] int id)
 
         return this.Redirect("~/slots/0");
     }
+
+    [HttpGet("flag")]
+    public async Task<IActionResult> FlagLevel([FromRoute] int id)
+    {
+        UserEntity? user = this.database.UserFromWebRequest(this.Request);
+        if (user == null) return this.StatusCode(403);
+
+        SlotEntity? slot = await this.database.Slots.FirstOrDefaultAsync(s => s.SlotId == id);
+        if (slot == null) return this.BadRequest();
+        if (slot.Creator == null) return this.StatusCode(500);
```

Comment:
This is kind of a redundant check here. a slot can't exist without a creator due to SQL constraints.